packages/@aws-cdk/tmp-toolkit-helpers/src/api/io/private/format/format-stack-diff.ts

```diff
@@ -0,0 +1,133 @@
+import type * as cxapi from '@aws-cdk/cx-api';
+import {
+  type DescribeChangeSetOutput,
+  formatDifferences,
+  fullDiff,
+  mangleLikeCloudFormation,
+} from '@aws-cdk/cloudformation-diff';
+import { buildLogicalToPathMap, logicalIdMapFromTemplate, obscureDiff, StringWriteStream } from './util';
+import { NestedStackTemplates } from './nested-stack-templates';
+
+/**
+ * Output of formatStackDiff
+ */
+export interface FormatStackDiffOutput {
+  /**
+   * Number of stacks with diff changes
+   */
+  readonly numStacksWithChanges: number;
+
+  /**
+   * Complete formatted diff
+   */
+  readonly formattedDiff: string;
+}
+
+/**
+ * Formats the differences between two template states and returns it as a string.
+ *
+ * @param oldTemplate the old/current state of the stack.
+ * @param newTemplate the new/target state of the stack.
+ * @param strict      do not filter out AWS::CDK::Metadata or Rules
+ * @param context     lines of context to use in arbitrary JSON diff
+ * @param quiet       silences \'There were no differences\' messages
+ *
+ * @returns the formatted diff, and the number of stacks in this stack tree that have differences, including the top-level root stack
+ */
+export function formatStackDiff(
+  oldTemplate: any,
+  newTemplate: cxapi.CloudFormationStackArtifact,
+  strict: boolean,
+  context: number,
+  quiet: boolean,
+  stackName?: string,
+  changeSet?: DescribeChangeSetOutput,
+  isImport?: boolean,
+  nestedStackTemplates?: { [nestedStackLogicalId: string]: NestedStackTemplates }): FormatStackDiffOutput {
+  let diff = fullDiff(oldTemplate, newTemplate.template, changeSet, isImport);
+
+  // The stack diff is formatted via `Formatter`, which takes in a stream
+  // and sends its output directly to that stream. To faciliate use of the
+  // global CliIoHost, we create our own stream to capture the output of
+  // `Formatter` and return the output as a string for the consumer of
+  // `formatStackDiff` to decide what to do with it.
+  const stream = new StringWriteStream();
+
+  let numStacksWithChanges = 0;
+  let formattedDiff = '';
+  let filteredChangesCount = 0;
+  try {
+    // must output the stack name if there are differences, even if quiet
+    if (stackName && (!quiet || !diff.isEmpty)) {
+      stream.write(format('Stack %s\n', chalk.bold(stackName)));
```

Comment:
welp, these prints don't work yet and im worried im not doing the right thing here: 

`format` and `chalk` don't exist in this context and i'm not sure how to replicate them.

Reply:
i guess a better question to ask was: are we certain we want it in tmp-toolkit-helpers and not imported [here](https://github.com/aws/aws-cdk-cli/blob/conroy/diffs/packages/@aws-cdk/toolkit-lib/lib/api/aws-cdk.ts) like a lot of other heavy duty imports.

but it sounds like your answer also answers that question too